lib/json/serialization.nit

```diff
@@ -448,6 +448,26 @@ class JsonDeserializer
 	end
 end
 
+redef class Text
+
+	# Deserialize a `nullable Object` from this JSON formatted string
+	#
+	# Warning: Deserialization errors are reported with `print_error` and
+	# may be returned as a partial object or as `null`.
+	#
+	# This method is not appropriate when errors need to be handled programmatically,
+	# manually use a `JsonDeserializer` in such cases.
+	fun from_json_string: nullable Object
+	do
+		var deserializer = new JsonDeserializer(self)
+		var res = deserializer.deserialize
+		if deserializer.errors.not_empty then
+			print_error "Deserialization Errors: {deserializer.errors.join(", ")}"
```

Comment:
It may not be the most common use case, but I had this precise problem in the program for which I originally wrote this code. The RESTful server returns instances of Error on server-side error, invalid requests and invalid user/pass combination. So the client expects to receive both Errors and other objects. 

Note that this problem also exists when deserializing `"null"`. However, even if our parser would deserialize such a string to `null`, it is not valid JSON.

The current API already preserves the errors quite efficiently but it may be intimidating to use. An alternative error-preserving API would save use at most 1 line compared to the recommended way to deal with errors:
~~~
var deserializer = new JsonDeserializer("JSON string...")
var res = deserializer.deserialize
if deserializer.errors.not_empty then
	print_error "Deserialization errors: {deserializer.errors.join(", ")}"
	return
end
if not res isa ExpectedObject then
	print_error "Deserialization error: Expected a `ExpectedObject`, got a {res?.class_name or else "null"}"
	return
end
# Work
~~~

This PR propose to avoid dealing with errors manually for scripts and prototypes, using:
~~~
var res = "JSON string...".from_json_string
if res isa ExpectedObject then
	# Work
else abort
~~~